### PR TITLE
Update YOLO example for the latest to support the latest TensorFlow version

### PR DESCRIPTION
**Pull request by JanuszL**
- fixes the issue with the latest TensorFlow version and YOLO
  example that results in
  `AttributeError: 'CollectiveAllReduce' object has no attribute '_pool'`.
  The issue comes from the workaround for
  https://github.com/tensorflow/tensorflow/issues/50487

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- fixes the issue with the latest TensorFlow version and YOLO
  example that results in
  `AttributeError: 'CollectiveAllReduce' object has no attribute '_pool'`.
  The issue comes from the workaround for
  https://github.com/tensorflow/tensorflow/issues/50487

### Affected modules and functionalities:
- docs/examples/use_cases/tensorflow/yolov4/src/train.py

### Key points relevant for the review:
- NA

### Tests:

- [x] Existing tests apply
  - TL3_YOLO_convergence
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: docs/examples/use_cases/tensorflow/yolov4/src/train.py [hunk 1]
```diff
@@ -112,7 +112,8 @@ def dataset_fn(input_context):
 
     multigpu = kwargs.get("multigpu")
     strategy = tf.distribute.MirroredStrategy() if multigpu else tf.distribute.get_strategy()
-    atexit.register(strategy._extended._collective_ops._pool.close) # type: ignore
+    if hasattr(strategy._extended._collective_ops, "_pool"):
+        atexit.register(strategy._extended._collective_ops._pool.close) # type: ignore
```
Comment (stiepan) [inline docs/examples/use_cases/tensorflow/yolov4/src/train.py:116]: Do we know if no WAR is needed in the new tf version where the old WAR does not work?

Comment (stiepan) [reply docs/examples/use_cases/tensorflow/yolov4/src/train.py:116]: I guess the passing tests may be the answer, but maybe it is random thingy?

Comment (JanuszL) [reply docs/examples/use_cases/tensorflow/yolov4/src/train.py:116]: 🤷 The CI will tell us.
Someone in https://github.com/tensorflow/tensorflow/issues/50487 claims that it works with the latest.

Comment (JanuszL) [reply docs/examples/use_cases/tensorflow/yolov4/src/train.py:116]: > maybe it is random thingy?

It was 100% reproducible before the fix.

**Review by stiepan — APPROVED** [review]

**Review by jantonguirao — APPROVED** [review]

**Pull request merged**